plugins/node/opentelemetry-instrumentation-dns/test/integrations/dnspromise-lookup.test.ts

```diff
@@ -173,16 +170,57 @@ describe('dns.promises.lookup()', () => {
         assert.strictEqual(spans.length, 1);
         assertSpan(span, {
           addresses: [],
-          // tslint:disable-next-line:no-any
-          hostname: hostname as any,
           forceStatus: {
             code: SpanStatusCode.ERROR,
             message: error!.message,
           },
         });
       }
     });
+
+    it('should omit dns.hostname attribute by default', async () => {
+      const hostname = 'google.com';
+      const config: DnsInstrumentationConfig = {};
+      instrumentation.setConfig(config);
+
+      try {
+        await lookupPromise(hostname as any, { family: 4 });
+        assert.fail();
+      } catch (error) {
+        const spans = memoryExporter.getFinishedSpans();
+        const [span] = spans;
+
+        assert.strictEqual(spans.length, 1);
+        assertSpan(span, {
+          addresses: [],
+          hostname: undefined,
```

Comment:
This check will not verify it's omitted, as the code in `assertSpan` will skip the check:
```js
  if (validations.hostname !== undefined) {
    assert.strictEqual(
      span.attributes[AttributeNames.DNS_HOSTNAME],
      validations.hostname
    );
  }
```

Maybe when the value sent to `assertSpan` is undefined we should check that it's not in attributes?

Reply:
fixed